web/client/epics/geoProcessingTools.js

```diff
@@ -148,44 +146,64 @@ export const checkWPSAvailabilityGPTEpic = (action$, store) => action$
                     return status && findIndex(ids, x => x === process) > -1;
                 }, true);
 
-                const actions = [
-                    setWPSAvailability(layerId, areAllWpsAvailable, source),
-                    checkingWPS(false)
-                ];
-                if (areAllWpsAvailable) {
-                    actions.push(getDescribeLayer(layerUrl, layer, {}, source));
+                if (!areAllWpsAvailable) {
+                    // just set the layer to invalid,
+                    // the wps flag to false
+                    // no need to ask the describe layer and describe feature type
+                    return Rx.Observable.from(
+                        [
+                            setInvalidLayer(layerId, source),
+                            setWPSAvailability(layerId, false, source),
+                            checkingWPS(false)
+                        ]);
+                }
+                // wps are all present, so we continue the checks
+                if (layer.describeFeatureType) {
+                    // we have the describe feature type available so we just fetch features
+                    return Rx.Observable.from([
+                        setWPSAvailability(layerId, true, source),
+                        getFeatures(layerId, source),
+                        checkingWPS(false)
+                    ]);
                 }
-                return Rx.Observable.from(actions);
+                // wps are all present, but
+                // we miss the describe feature type info so we trigger it and we wait for the answer
+
+                return Rx.Observable.from([
+                    setWPSAvailability(layerId, true, source),
+                    getDescribeLayer(layerUrl, layer, {}, source)
+                ])
+                    .merge(
+                        Rx.Observable.race(
+                            action$.ofType(UPDATE_NODE).filter(({node}) => node === layerId),
+                            Rx.Observable.timer(10 * 1000).mapTo("timeout")
+                        )
+                            .switchMap((act) => {
+                                // if the describe ft goes in timeout or has an error we invalidate the layer
+                                if (act === "timeout" || act?.error) {
+                                    return Rx.Observable.of(setInvalidLayer(layerId, source));
```

Comment:
In this case ` checkingWPS(false)` is not called. I suggest to put it outside, so you don't have to care about it anymore.

Reply:
right, but it was not working put outside the root function

i can move it after the switch map

Reply:
If you move it at line 199
```
.startWith(checkingWPS(true)).concat(Rx.Obserrvable.of(checkingWPS(false)))
```
I think that you can remove it from all the other points and forget to handle it in every single branch of your workflow.

Reply:
i was thinking the same, but no it's not working if i place it there 
i tested it

also consider that this checkingWPS should be trigger as soon it is completed not after describe layer etc and the whole flow

Reply:
Notice that in the solution you proposed `checkingWPS`  it is triggered after the describeLayer is completed: 
```
switchMap((act) => {
                                // if the describe ft goes in timeout or has an error we invalidate the layer
                                if (act === "timeout" || act?.error) {
                                    return Rx.Observable.of(setInvalidLayer(layerId, source));
                                }
                                return Rx.Observable.of(
                                    getFeatures(layerId, source));
                            }).concat([checkingWPS(false)])
```

So if you say :

> this checkingWPS should be trigger as soon it is completed  **not** after describe layer"

So you have to change something.

Reply:
changed again